salt/pillar/nsot.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+'''
+A module that adds data to the Pillar structure from the NSoT API.
+
+.. versionadded:: Neon
+
+Configuring the NSoT ext_pillar
+---------------------------------
+
+The following fields are required:
+
+.. code-block:: yaml
+
+    ext_pillar:
+      - nsot:
+        api_url: http://nsot_url.com/api/
+        email: 'user@site.com'
+        secret_key: abc123
+
+.. note::
+
+    Note the trailing slash on the api_url field. Although NSoT gives the option
+    of whether or not to use the tailing slash, this module assumes that the
+    APPEND_SLASH option is set to True (the default)
+
+The following fields are optional:
+
+replace_dot: ``str``
+    This is in case a minion_id contains a dot. Because NSoT
+    doesn't allow hostnmes with dots, this option allows you to look up a device
+    with this character replacing dots. For example, with this option set to
+     '-', the minion rtr1.lax1 would be looked up in NSoT as rtr1-lax1.
+
+all_device_info: ``list``
+    This is a list of minions that will receive all device
+    information in NSoT. For example, if this list contained the entry
+    'test_min', then the minion 'test_min' would have all devices in NSoT under
+    pillar['nsot']['devices']
```

Comment:
How to give this to my 5K machines? This should support the same regexes as `minion_regex`.

Reply:
Fair point. I will change this.